openapi3/operation.go

```
@@ -162,24 +162,18 @@ func (operation Operation) JSONLookup(token string) (interface{}, error) {
 }
 
 func (operation *Operation) AddParameter(p *Parameter) {
-	operation.Parameters = append(operation.Parameters, &ParameterRef{
-		Value: p,
-	})
+	operation.Parameters = append(operation.Parameters, &ParameterRef{Value: p})
 }
 
 func (operation *Operation) AddResponse(status int, response *Response) {
-	responses := operation.Responses
-	if responses == nil {
-		responses = NewResponses()
-		operation.Responses = responses
-	}
 	code := "default"
-	if status != 0 {
+	if 0 < status && status < 1000 {
 		code = strconv.FormatInt(int64(status), 10)
 	}
-	responses[code] = &ResponseRef{
-		Value: response,
+	if operation.Responses == nil {
+		operation.Responses = &Responses{}
 	}
+	operation.Responses.Set(code, &ResponseRef{Value: response})
```

`Responses.Set(..)` should fail loudly if responses is nil